Prog/Crimes_forweb.sas

```diff
@@ -0,0 +1,86 @@
+/**************************************************************************
+ Program:  Crime_forweb
+ Library:  Police
+ Project:  NeighborhoodInfo DC
+ Author:   Rob Pitingolo
+ Created:  11/07/2017
+ Version:  SAS 9.4
+ Environment:  Windows
+ Modifications: 
+
+**************************************************************************/
+
+%include "L:\SAS\Inc\StdLocal.sas"; 
+
+** Define libraries **;
+%DCData_lib( Police )
+%DCData_lib( Web )
+
+
+/***** Update the let statements for the data you want to create CSV files for *****/
+
+%let library = police; /* Library of the summary data to be transposed */
+%let outfolder = crime; /* Name of folder where output CSV will be saved */
+%let sumdata = crimes_sum; /* Summary dataset name (without geo suffix) */
+%let start = 2000; /* Start year */
+%let end = 2016; /* End year */
+%let keepvars = crimes_pt1 Crimes_pt1_property Crimes_pt1_violent Crime_rate_pop; /* Summary variables to keep and transpose */
+
+
+/***** Update the web_varcreate marcro if you need to create final indicators for the website after transposing *****/
+
+%macro web_varcreate;
+
+Crimes_pt1_per1000 = crimes_pt1 / Crime_rate_pop*1000;
+Crimes_pt1_property_per1000 = Crimes_pt1_property / Crime_rate_pop*1000;
+Crimes_pt1_violent_per1000 = Crimes_pt1_violent / Crime_rate_pop*1000;
+
+%mend web_varcreate;
+
+
+
+/**************** DO NOT UPDATE BELOW THIS LINE ****************/
+
+%macro csv_create(geo);
+			 
+%web_transpose(&library., &outfolder., &sumdata., &geo., &start., &end., &keepvars. );
+
+/* Load transposed data, create indicators for profiles */
+data &sumdata._&geo._long_allyr;
+	set &sumdata._&geo._long;
+	%web_varcreate;
+run;
+
+/* Output all-years CSV */
+ods csv file ="&_dcdata_default_path.\web\output\&outfolder.\&geo.\&outfolder._&geo._AllYears.csv";
+	proc print data = &sumdata._&geo._long_allyr noobs;
+	run;
+ods csv close;
+
+
+/*Subset datasets by year */
+%do j = &start. %to &end.;
+data &sumdata._&geo._long_&j. ;
+	set &sumdata._&geo._long_allyr;
+	where timeframe ="&j." ;
+run;
+%runquit;
```

Comment:
The definition of the _%runquit_ macro seems to be embedded in file for _%web_transpose()_. Since this macro is being used outside of _%web_transpose()_, it should be defined in it's own autocall macro file (i.e., **runquit.sas**).  (General rule is that each autocall macro file should include the definition for just one macro. But, if a "sub" macro is used exclusively within the execution of an autocall macro, I think it's OK to include the definition in the main macro file.)